### PR TITLE
[PUBDEV-4822] Fix external backend utilities

### PR DESCRIPTION
**Pull request by mmalohlava**
Spark vectors were not properly written into H2OFrame.

### CODE REVIEW

**Comment by mmalohlava** [comment]
@h2o-ops please test!

File: h2o-core/src/test/java/water/ExternalFrameWriterClientTest.java [hunk 1]
```diff
@@ -148,4 +112,192 @@ public void run() {
...
+                        op.doWrite(writer);
+
+                        try {
+                            writer.waitUntilAllWritten(10);
+                        } catch (ExternalFrameConfirmationException e){
+                            e.printStackTrace();
+                        }
+                        sock.close();
+                        rowsPerChunk[currentIndex] = op.nrows();
+                    } catch (IOException ignore) {
```
Comment (michalkurka) [inline h2o-core/src/test/java/water/ExternalFrameWriterClientTest.java]: Why are the exception ignored?

File: h2o-core/src/test/java/water/ExternalFrameWriterClientTest.java [hunk 2]
```diff
@@ -27,97 +28,60 @@ public static void setup() {
...
+        // we will open 2 connection per h2o node
+        final String[] connStrings = ArrayUtils.join(nodes, nodes);
 
-        Frame frame = null;
+        Frame frame = createFrame(testOp, connStrings);
         try {
-            frame = DKV.getGet(frameName);
             assertEquals(frame.anyVec().nChunks(), connStrings.length);
             assertEquals(frame._names.length, 4);
             assertEquals(frame.numCols(), 4);
```
Comment (michalkurka) [inline h2o-core/src/test/java/water/ExternalFrameWriterClientTest.java]: I don't like that it is only checking the values that are filled. Ideally we should check the whole frame.

Also the comment "// last row should be NA" is misplaced, it should ideally be on line 104 of the original source file.

Comment (mmalohlava) [inline h2o-core/src/test/java/water/ExternalFrameWriterClientTest.java]: yup, agree, i missed that in original code.

**Comment by mmalohlava** [comment]
@h2o-ops please test!

File: h2o-core/src/main/java/water/ExternalFrameUtils.java [hunk 3]
```diff
@@ -54,7 +54,7 @@ public static ByteChannel getConnection(String ipPort) throws IOException{
 
     public static byte[] vecTypesFromExpectedTypes(byte[] expectedTypes, int[] vecElemSizes){
         int size = expectedTypes.length;
-        if(vecElemSizes != null && vecElemSizes.length == 0){
+        if(vecElemSizes != null && vecElemSizes.length > 0){
             size = size - vecElemSizes.length;
```
Comment (mmalohlava) [inline h2o-core/src/main/java/water/ExternalFrameUtils.java]: Ups, this is still wrong. Mea culpa

**Pull request merged**